Github Friend addition, deletion, and friend list integration with the new MongoDB URL

Github Friend addition, deletion, and friend list integration with the new MongoDB URL

## Description
The friend's table has been migrated to the no-SQL database, and so some changes with the url have been done. Features like add friend, delete friend, and get friend list are not working because of this.

## Expected Behavior
When clicked, the friend's username should be added

## Actual Behavior
Error pops up
## Possible Fix
Changing the URL in the front end and changing the serializer would work. You can see commit 162 for help

D:\Open Source\Leaderboard-Pro\api>make migrate
python3 manage.py makemigrations
Traceback (most recent call last):
  File "D:\Open Source\Leaderboard-Pro\api\manage.py", line 22, in <module>
    main()
  File "D:\Open Source\Leaderboard-Pro\api\manage.py", line 18, in main
    execute_from_command_line(sys.argv)
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\core\management\__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\core\management\__init__.py", line 395, in execute
    django.setup()
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\apps\registry.py", line 114, in populate
    app_config.import_models()
  File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\apps\config.py", line 301, in import_models
    self.models_module = import_module(models_module_name)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1520.0_x64__qbz5n2kfra8p0\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 940, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "D:\Open Source\Leaderboard-Pro\api\leaderboard\models.py", line 50, in <module>
    class codeforcesUser(models.Model):
  File "D:\Open Source\Leaderboard-Pro\api\leaderboard\models.py", line 55, in codeforcesUser
    default=datetime.max.timestamp()
            ^^^^^^^^^^^^^^^^^^^^^^^^
OSError: [Errno 22] Invalid argument
make: *** [Makefile:5: migrate] Error 1

> D:\Open Source\Leaderboard-Pro\api>make migrate python3 manage.py makemigrations Traceback (most recent call last): File "D:\Open Source\Leaderboard-Pro\api\manage.py", line 22, in main() File "D:\Open Source\Leaderboard-Pro\api\manage.py", line 18, in main execute_from_command_line(sys.argv) File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\core\management__init__.py", line 419, in execute_from_command_line utility.execute() File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\core\management__init__.py", line 395, in execute django.setup() File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django__init__.py", line 24, in setup apps.populate(settings.INSTALLED_APPS) File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\apps\registry.py", line 114, in populate app_config.import_models() File "C:\Users\HP\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\django\apps\config.py", line 301, in import_models self.models_module = import_module(models_module_name) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.1520.0_x64__qbz5n2kfra8p0\Lib\importlib__init__.py", line 126, in import_module return _bootstrap._gcd_import(name[level:], package, level) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "", line 1204, in _gcd_import File "", line 1176, in _find_and_load File "", line 1147, in _find_and_load_unlocked File "", line 690, in _load_unlocked File "", line 940, in exec_module File "", line 241, in _call_with_frames_removed File "D:\Open Source\Leaderboard-Pro\api\leaderboard\models.py", line 50, in class codeforcesUser(models.Model): File "D:\Open Source\Leaderboard-Pro\api\leaderboard\models.py", line 55, in codeforcesUser default=datetime.max.timestamp() ^^^^^^^^^^^^^^^^^^^^^^^^ OSError: [Errno 22] Invalid argument make: *** [Makefile:5: migrate] Error 1

Just change the following lines in views.py
line 4 -from datetime import datetime
line 44 default=datetime.max.timestamp
Some Windows user faces this problem due to some dependency clashes.
Save and again try migrating the database. Please use your custom database for testing

--- a/app/src/components/GithubTable.js
+++ b/app/src/components/GithubTable.js
@@ -41,7 +41,7 @@ export const GithubTable = ({
   const [filteredusers, setFilteredusers] = useState([]);
   const [todisplayusers, setTodisplayusers] = useState([]);
   const getghfriends = async () => {
-    const response = await fetch("http://localhost:8000/api/getghfriends/", {
+    const response = await fetch("http://127.0.0.1:8000/githubFL/", {
       method: "GET",
       headers: {
         "Content-Type": "application/json",
@@ -57,8 +57,7 @@ export const GithubTable = ({
   };
 
   async function addfriend(e) {
-    
-    const response = await fetch("http://localhost:8000/api/ghfriends/", {
+    const response = await fetch("http://127.0.0.1:8000/githubFA/", {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
@@ -66,7 +65,7 @@ export const GithubTable = ({
           "Bearer " + JSON.parse(localStorage.getItem("authTokens")).access,
       },
       body: JSON.stringify({
-        ghFriend_uname: e.username,
+        friendName: e.username,
       }),
     });
     if (response.status !== 200) {
@@ -78,8 +77,7 @@ export const GithubTable = ({
     }
   }
   async function dropfriend(e) {
-    
-    const response = await fetch("http://localhost:8000/api/dropghfriends/", {
+    const response = await fetch("http://127.0.0.1:8000/githubFD/", {
       method: "POST",
       headers: {
         "Content-Type": "application/json",
@@ -87,7 +85,7 @@ export const GithubTable = ({
           "Bearer " + JSON.parse(localStorage.getItem("authTokens")).access,
       },
       body: JSON.stringify({
-        ghFriend_uname: e,
+        friendName: e,
       }),
     });
     if (response.status !== 200) {

